homeassistant/components/mqtt/device_trigger.py

```diff
@@ -0,0 +1,210 @@
+"""Provides device automations for MQTT."""
+import logging
+from typing import List
+
+import attr
+import voluptuous as vol
+
+from homeassistant.components import mqtt
+from homeassistant.components.automation import AutomationActionType
+import homeassistant.components.automation.mqtt as automation_mqtt
+from homeassistant.components.device_automation import TRIGGER_BASE_SCHEMA
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_PLATFORM, CONF_TYPE
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.typing import ConfigType
+
+from . import (
+    ATTR_DISCOVERY_HASH,
+    CONF_CONNECTIONS,
+    CONF_DEVICE,
+    CONF_IDENTIFIERS,
+    CONF_PAYLOAD,
+    CONF_QOS,
+    DOMAIN,
+)
+from .const import DEFAULT_QOS
+from .discovery import MQTT_DISCOVERY_UPDATED, clear_discovery_hash
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_AUTOMATION_TYPE = "automation_type"
+CONF_SUBTYPE = "subtype"
+CONF_TOPIC = "topic"
+DEFAULT_ENCODING = "utf-8"
+DEVICE = "device"
+
+MQTT_TRIGGER = {
+    # Trigger when MQTT message is received
+    CONF_PLATFORM: DEVICE,
+    CONF_DOMAIN: DOMAIN,
+}
+
+TRIGGER_SCHEMA = TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_PLATFORM): DEVICE,
+        vol.Required(CONF_DOMAIN): DOMAIN,
+        vol.Optional(CONF_DEVICE_ID): str,
+        vol.Required(CONF_TOPIC): mqtt.valid_subscribe_topic,
+        vol.Optional(CONF_PAYLOAD, default=None): vol.Any(None, cv.string),
+        vol.Optional(CONF_QOS, default=DEFAULT_QOS): vol.Coerce(int),
+        vol.Required(CONF_TYPE): cv.string,
+        vol.Required(CONF_SUBTYPE): cv.string,
+    }
+)
+
+PLATFORM_SCHEMA = mqtt.MQTT_BASE_PLATFORM_SCHEMA.extend(
```

Comment:
What does platform mean here ?

Reply:
What would be a better name, "CONFIG_SCHEMA" maybe?

Reply:
Name it after what it is validating. So I guess it's not trigger schema… so what is it ?

Reply:
Renamed, I think it's clearer now.